### PR TITLE
DD4hep: Add Reflection Rotation

### PR DESCRIPTION
**Pull request by ianna**
* Add ReflectionRotation tag processing
* Add a Trd2 division (though Trd1 is needed) - see issue https://github.com/AIDASoft/DD4hep/issues/460
* Make sure a default namespace is added to locally referred names
* Update Muon test configuration: it's fully parsed*)

*) See the note about Trd1

### CODE REVIEW

**Comment by ianna** [comment]
please test

**Comment by ianna** [comment]
+1

**Comment by fabiocos** [comment]
+1

File: DetectorDescription/DDCMS/src/DDAlgoArguments.cc [hunk 1]
```diff
@@ -33,6 +33,29 @@ cms::makeRotation3D( double thetaX, double phiX,
...
+		     double thetaY, double phiY,
+		     double thetaZ, double phiZ ) {
+  // define 3 unit std::vectors forming the new left-handed axes
+  DD3Vector x( cos( phiX ) * sin( thetaX ), sin( phiX ) * sin( thetaX ), cos( thetaX ));
+  DD3Vector y( cos( phiY ) * sin( thetaY ), sin( phiY ) * sin( thetaY ), cos( thetaY ));
+  DD3Vector z( cos( phiZ ) * sin( thetaZ ), sin( phiZ ) * sin( thetaZ ), cos( thetaZ ));
+
+  double tol = 1.0e-3; // Geant4 compatible
+  double check = ( x.Cross( y )).Dot( z ); // in case of a LEFT-handed orthogonal system this must be -1
+  if( fabs( 1. + check ) > tol ) {
```
Comment (cvuosalo) [inline DetectorDescription/DDCMS/src/DDAlgoArguments.cc]: I think we should be using std::abs not fabs.

File: DetectorDescription/DDCMS/src/DDAlgoArguments.cc [hunk 2]
```diff
@@ -33,6 +33,29 @@ cms::makeRotation3D( double thetaX, double phiX,
...
+dd4hep::Rotation3D
+cms::makeRotReflect( double thetaX, double phiX,
+		     double thetaY, double phiY,
+		     double thetaZ, double phiZ ) {
+  // define 3 unit std::vectors forming the new left-handed axes
+  DD3Vector x( cos( phiX ) * sin( thetaX ), sin( phiX ) * sin( thetaX ), cos( thetaX ));
+  DD3Vector y( cos( phiY ) * sin( thetaY ), sin( phiY ) * sin( thetaY ), cos( thetaY ));
+  DD3Vector z( cos( phiZ ) * sin( thetaZ ), sin( phiZ ) * sin( thetaZ ), cos( thetaZ ));
+
+  double tol = 1.0e-3; // Geant4 compatible
```
Comment (cvuosalo) [inline DetectorDescription/DDCMS/src/DDAlgoArguments.cc]: constexpr double tol

**Pull request merged**